src/oidc_child/oidc_child_json.c

```diff
@@ -0,0 +1,505 @@
+
+/*
+    SSSD
+
+    Helper child for OIDC and OAuth 2.0 Device Authorization Grant
+    JSON utilities
+
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+
+    Copyright (C) 2022 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <jose/jws.h>
+#include <jose/b64.h>
+#include <jansson.h>
+
+#include "oidc_child/oidc_child_util.h"
+
+static char *get_json_string(TALLOC_CTX *mem_ctx, json_t *root,
```

Comment:
`root` could be a const pointer?

Reply:
fixed

Reply:
Sorry I didn't mention explicitly, but the same applies to `get_json_integer()` and `get_json_scope()`

Reply:
fixed